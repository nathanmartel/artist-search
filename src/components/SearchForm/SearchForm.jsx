import React from 'react';
import PropTypes from 'prop-types';
import './SearchForm.css';

const SearchForm = ({ query = '', onQueryChange, onSubmit }) => {
  return (
    <form onSubmit={onSubmit}>
      <input type="text" value={query} onChange={onQueryChange} />
      <button onSubmit={() => onSubmit}>Submit</button>
    </form>
  );
};

SearchForm.propTypes = {
  query: PropTypes.string.isRequired,
  onQueryChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default SearchForm;
