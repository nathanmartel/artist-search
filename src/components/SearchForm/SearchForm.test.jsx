import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm component', () => {
  it('renders SearchForm', () => {
    const wrapper = shallow(<SearchForm artist={'Radiohead'} onArtistChange={() => {}} onSubmit={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
