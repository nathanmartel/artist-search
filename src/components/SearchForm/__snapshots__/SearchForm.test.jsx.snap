// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders SearchForm 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="Radiohead"
  />
  <button
    onSubmit={[Function]}
  >
    Submit
  </button>
</form>
`;
